### ISSUE TITLE
Doc test accepts semicolon struct definition with braces

### ISSUE THREAD

**Issue by rasmuspeders1** [rust-lang/rust#113362]
This doc test code compiles fine even though the code would not be accepted normally. Notice the semicolon in `struct Example{};` this semicolon is not allowed in the actual code but is accepted in the doc example.

```
/// # Example
/// ```
/// struct Example{};
/// ```
struct Example{}
```
I would have expected a compile error when running the `cargo test` command.

Instead it compiles and runs the doc example without problems.

### Meta

`rustc --version --verbose`:
```
rustc 1.72.0-nightly (839e9a6e1 2023-07-02)
binary: rustc
commit-hash: 839e9a6e1210934fd24b15548b811a97c77138fc
commit-date: 2023-07-02
host: x86_64-unknown-linux-gnu
release: 1.72.0-nightly
LLVM version: 16.0.5
```

**Comment by rasmuspeders1** [comment]
Sorry if this is a potential duplicate or if this is expected behavior.
I found issue https://github.com/rust-lang/rust/issues/91398 which indicates that rust doc did indeed handle this as I would expect at some point earlier.

**Comment by lukas-code** [comment]
I think this is expected behavior. Because the doctest contains no main function, it gets wrapped in an implicit main function and this is perfectly valid rust code:

```rust
fn main() {
    struct Example{};
}
```

It's an item statement, followed by an empty statement.

The issue you linked fails to compile, because its doctest does contain a main function and is therefore not wrapped in another main function.

Sources:
* [The Rust Reference on Statements](https://doc.rust-lang.org/nightly/reference/statements.html)
* [The rustdoc book on pre-processing examples](https://doc.rust-lang.org/nightly/rustdoc/write-documentation/documentation-tests.html#pre-processing-examples)

**Comment by rasmuspeders1** [comment]
I see now.  Thanks.
I was expecting to be able to rely on the doc test code always  compiling if copy pasted and compiled as a standalone program. 
I can see how I cannot always rely on this and how it is not something that can(or maybe even should) be fixed easily.
